libr/bin/p/bin_dyldcache.c

```diff
@@ -828,35 +828,41 @@ static RList *create_cache_bins(RBinFile *bf, RBuffer *cache_buf, cache_hdr_t *h
 			return NULL;
 		}
 
-		ut16 *depArray = R_NEWS0 (ut16, accel->depListCount);
-		if (!depArray) {
-			r_list_free (target_lib_names);
-			R_FREE (target_libs);
-			r_list_free (bins);
-			R_FREE (deps);
-			R_FREE (img);
-			return NULL;
-		}
+		ut16 *depArray = NULL;
+		cache_imgxtr_t *extras = NULL;
+		if (accel) {
+			depArray = R_NEWS0 (ut16, accel->depListCount);
+			if (!depArray) {
+				r_list_free (target_lib_names);
```

Comment:
better use goto, thise collection of frees can be written once at the bottom of the function